crates/partition-store/src/partition_store_manager.rs

```diff
@@ -194,6 +198,52 @@ impl PartitionStoreManager {
         Ok(partition_store)
     }
 
+    pub async fn export_partition_snapshot(
+        &self,
+        partition_id: PartitionId,
+        snapshot_id: SnapshotId,
+        snapshot_base_path: PathBuf,
+    ) -> anyhow::Result<LocalPartitionSnapshot> {
+        let mut guard = self.lookup.lock().await;
+
+        let partition_store = guard
+            .live
+            .get_mut(&partition_id)
+            .ok_or(anyhow!("Unknown partition: {}", partition_id))?;
```

Comment:
I don't see another way to grab a `partition_store` here? This would have the effect of serializing snapshot generation if we get a bunch of them at once, which is not a bad thing as I see it. There are no other contenders for this lock at runtime currently, only on startup.